### PR TITLE
Separate View into different implementations

### PR DESCRIPTION
**Pull request by thewilkybarkid**
#56 gave rise to the idea of having a lazy `View`. That experiment hacked it in; but it's better to have a real proxy implementation. After a bit of thought, we _already_ have the need for different implementations (primarily to stop simple strings having to be rendered as a template). So, after a bit of a refactor in https://github.com/libero/pattern-library/pull/76 we can have a separate `StringView` to handle that case.

### CODE REVIEW

**Comment by thewilkybarkid** [comment]
As pointed out by @nlisgo, doesn't actually rely on https://github.com/libero/pattern-library/pull/76 so we can separate them.

File: vendor-extra/ViewsBundle/src/Views/TemplateView.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    public function withContext(array $context) : TemplateView
+    {
+        $view = clone $this;
+
+        $view->context = array_replace_recursive($view->context, $context);
+
+        return $view;
+    }
+
+    public function jsonSerialize()
```
Comment (giorgiosironi) [inline vendor-extra/ViewsBundle/src/Views/TemplateView.php]: this is not serializing to JSON (a string conforming to https://json.org/); it's serializing to a PHP data structure that can be encoded in JSON if needed (associative array, string, null)

Comment (giorgiosironi) [reply vendor-extra/ViewsBundle/src/Views/TemplateView.php]: But it's a PHP standard so imposed

Comment (giorgiosironi) [reply vendor-extra/ViewsBundle/src/Views/TemplateView.php]: Why are we serializing to JSON? Only debugging comes to mind, as an equivalent of `var_dump()`/`var_export()`

Comment (thewilkybarkid) [reply vendor-extra/ViewsBundle/src/Views/TemplateView.php]: Yeah, it was useful when debugging. Since Xdebug has been set up, won't be needed.

**Comment by thewilkybarkid** [comment]
Bit of refactoring done with #64 in mind.

**Comment by nlisgo** [comment]
Happy with these changes. It is a useful distinction.

**Comment by thewilkybarkid** [comment]
Any more comments @giorgiosironi?

**Pull request merged**